test/next_ls/helpers/ast_helpers_variables_test.exs

```diff
@@ -0,0 +1,338 @@
+    end
+
+    test "symbol set in match with 'if' containing it's shadow", %{source: source} do
+      refs = Variables.list_variable_references(source, {13, 5})
+      assert length(refs) == 1
+      assert exists_in?(refs, {:bravo, {24..24, 18..23}})
+    end
+
+    test "symbol set in 'case' clause", %{source: source} do
+      refs = Variables.list_variable_references(source, {33, 13})
+      assert length(refs) == 2
+      assert exists_in?(refs, {:bravo, {33..33, 32..37}})
+      assert exists_in?(refs, {:bravo, {34..34, 19..24}})
+    end
+
+    test "symbol referenced in 'cond' clause", %{source: source} do
+      refs = Variables.list_variable_references(source, {46, 5})
+      assert length(refs) == 2
+      assert exists_in?(refs, {:alpha, {51..51, 9..14}})
+      assert exists_in?(refs, {:alpha, {58..58, 16..21}})
+    end
+
+    test "symbol shadowed in 'for' and 'with'", %{source: source} do
+      refs = Variables.list_variable_references(source, {62, 12})
+      assert length(refs) == 4
+      assert exists_in?(refs, {:alpha, {65..65, 18..23}})
+      assert exists_in?(refs, {:alpha, {69..69, 18..23}})
+      assert exists_in?(refs, {:alpha, {73..73, 26..31}})
+      assert exists_in?(refs, {:alpha, {81..81, 18..23}})
+
+      refs2 = Variables.list_variable_references(source, {63, 5})
+      assert length(refs2) == 1
+      assert exists_in?(refs2, {:bravo, {81..81, 11..16}})
+    end
+
+    test "symbol defined in 'for'", %{source: source} do
+      refs = Variables.list_variable_references(source, {65, 9})
+      assert length(refs) == 1
+      assert exists_in?(refs, {:alpha, {66..66, 18..23}})
+
+      refs2 = Variables.list_variable_references(source, {69, 9})
+      assert length(refs2) == 1
+      assert exists_in?(refs2, {:bravo, {69..69, 47..52}})
+
+      refs3 = Variables.list_variable_references(source, {69, 25})
+      assert length(refs3) == 2
+      assert exists_in?(refs3, {:charlie, {69..69, 55..62}})
+      assert exists_in?(refs3, {:charlie, {70..70, 18..25}})
+    end
+
+    test "symbol defined in 'with'", %{source: source} do
+      refs = Variables.list_variable_references(source, {73, 16})
+      assert length(refs) == 1
+      assert exists_in?(refs, {:delta, {74..74, 27..32}})
+
+      refs2 = Variables.list_variable_references(source, {74, 11})
+      assert length(refs2) == 1
+      assert exists_in?(refs2, {:delta, {75..75, 18..23}})
+
+      refs3 = Variables.list_variable_references(source, {78, 7})
+      assert length(refs3) == 1
+      assert exists_in?(refs3, {:error, {78..78, 25..30}})
+    end
+
+    test "symbol shadowed by anonymus funciton", %{source: source} do
+      refs = Variables.list_variable_references(source, {84, 12})
+      assert length(refs) == 2
+      assert exists_in?(refs, {:alpha, {96..96, 7..12}})
+      assert exists_in?(refs, {:alpha, {100..100, 21..26}})
+    end
+
+    test "symbol defined in anonymus funciton", %{source: source} do
```

Comment:
```suggestion
    test "symbol defined in anonymous funciton", %{source: source} do
```

Reply:
sorry for spelling, should add spellchecker